src/TikaOnDotnet.TextExtractor/TextExtractorOCR.cs

```diff
@@ -0,0 +1,167 @@
+            set { _tesseractOcrConfig = value; }
+        }
+
+        /// <summary>
+        /// Set to <c>true</c> to extract text from scanned PDF's with the use of Tesseract.
+        /// When set to <c>false</c> (default) then images in PDF's are ignored
+        /// </summary>
+        /// <remarks>
+        /// Tesseract has to be setup trough the <see cref="TesseractOCRConfig"/> property to
+        /// make this work. When not set then this property is ignored
+        /// </remarks>
+        public bool ExtractFromScannedPdfs { get; set; }
+        #endregion
+
+        #region Extract
+        /// <summary>
+        ///     Extract the text from the given inputstreams and returns it as an <see cref="TextExtractionResult" /> object
+        /// </summary>
+        /// <param name="streamFactory"></param>
+        /// <returns></returns>
+        /// <exception cref="TextExtractionException"></exception>
+        public new TextExtractionResult Extract(Func<Metadata, InputStream> streamFactory)
+        {
+            try
+            {
+                var parser = new AutoDetectParser();
+                var metadata = new Metadata();
+                //var outputWriter = new StringWriter();
+                var parseContext = new ParseContext();
+
+                if (_tesseractOcrConfig != null)
+                {
+                    parseContext.set(typeof (TesseractOCRConfig), _tesseractOcrConfig);
+
+                    if (ExtractFromScannedPdfs)
+                    {
+                        var pdfParserConfig = new PDFParserConfig();
+                        pdfParserConfig.setExtractInlineImages(true);
+                        // Set to false if pdf contains multiple images
+                        pdfParserConfig.setExtractUniqueInlineImagesOnly(false);
+                        parseContext.set(typeof (PDFParserConfig), pdfParserConfig);
+                    }
+                }
+
+                // Use the base class type for the key or parts of Tika won't find a usable parser
+                parseContext.set(typeof (Parser), parser);
+
+                var content = new System.IO.StringWriter();
+                var customTextContentHandler = new CustomTextContentHandler(content);
+
+                using (var inputStream = streamFactory(metadata))
+                {
+                    try
+                    {
+                        parser.parse(inputStream, customTextContentHandler, metadata, parseContext);
+                    }
+                    finally
+                    {
+                        inputStream.close();
+                    }
+                }
+
+                return Helpers.AssembleExtractionResult(content.ToString(), metadata);
+            }
+            catch (Exception ex)
+            {
+                throw new TextExtractionException("Extraction failed.", ex);
+            }
+        }
+        #endregion
+
+        #region TikaConfigDump
```

Comment:
Not sure what this is for. What do you think about moving this to be an extension method on `TikaConfig`?